src/lib/RequestParser.js

```diff
@@ -78,13 +79,16 @@ class RequestParser { // eslint-disable-line no-unused-vars
 
     /**
      * @param {any} keyId
+     * @param {boolean} [useLegacyStore = false]
      * @returns {Promise<KeyInfo>}
      */
-    async parseKeyId(keyId) {
+    async parseKeyId(keyId, useLegacyStore = false) {
```

Comment:
Could this potentially be determined by the used `keyId`? As far as I can tell the `AccountStore` used the `userFriendlyAddress` as key while the `KeyStore` uses `id`.